cdap-validation/bin/testing.py

```diff
@@ -0,0 +1,242 @@
+
+    # validate_connection
+    if cluster_vars['verbose'] == 2:  print 'Validating connection'
+    api_tests = [{'cm': base_vars['cm']['api_test']},{'ambari': base_vars['ambari']['api_test']}]
+    version_test = validate_connection(base_vars, cluster_vars, api_tests)
+    if cluster_vars['verbose'] == 2:  print ''
+    if cluster_vars['verbose'] == 2:  print 'version_test=%s' % (version_test)
+
+    # now we determine the install manager from the info we have
+    if cluster_vars['verbose'] == 2:  print 'Determine Hadoop install manager'
+    install_manager = get_install_manager(version_test)
+    cluster_vars['manager'] = install_manager
+    if cluster_vars['verbose'] == 2:  print 'Hadoop install manager is %s' % (install_manager)
+
+    # get version for API call
+    if install_manager == 'cm':
+        cluster_vars['version'] = version_test
+    else:
+        cluster_vars['version'] = 'v1'
+    
+    cluster_vars['base_url'] = cluster_vars['host'] + cluster_vars['version']
+    
+    # get preliminary info
+    cluster_vars = get_install_manager_info(base_vars, cluster_vars, install_manager)
+      # do we need these (below) or can we just work with the dict values?
+    if cluster_vars['verbose'] == 2:
+        for key, value in cluster_vars.iteritems():
+            print "%s=%s" % (key, value)
+
+    # get and run commands (API commands for now)
+    get_commands(base_vars, cluster_vars)
+
+    #########################################
+
+    ##### MODULES #####
+    
+    # find_modules
+    # navigate through ./module/* directories and look for module.json files
+    # parse through those to determine modules that will be run
+    # save (follow design)
+
+    if cluster_vars['verbose'] == 2:  print 'verbose=%s' % (cluster_vars['verbose'])
+    if cluster_vars['verbose'] == 2:  print '\nfind modules:\n'
+
+    module_list = find_modules('module.json', 'modules')
+    if cluster_vars['verbose'] == 2:
+        for module in module_list:
+            print 'module = %s' % (module)
+
+    modules = AutoVivification()
+    module_name_list = []
+   
+    for module in module_list:
+        m = re.search('/(.+?)/', module)
+        if m:
+            module_name = m.group(1)
+            modules[module_name] = create_module_json(module_list)
+            module_name_list.append(module_name)
+
+    if cluster_vars['verbose'] == 2: print modules
+
+    # run modules
+    # execute modules: for every known module, run specific functions (see module_functions)
+    if cluster_vars['verbose'] == 2: print '\nrun modules:\n'
+    run_modules(base_vars, cluster_vars, modules, module_name_list, install_manager) 
+    # send params necessary for module to run
+    # e.g. config_validator needs to know location of the base ref and stored results files
+
+###########################################################################################
+# reads JSON object passed to the framework
+def read_input(input_file):
```

Comment:
read_file_as_json would be more descriptive. Otherwise you don't know what this function takes and what it returns